### PR TITLE
IS-707: Make CI fail on warnings

### PR DESCRIPTION
**Pull request by stevengubler**
# [IS-707](https://jira.hyperledger.org/browse/IS-707)

To be in compliance with CII certification standards, we must address all warnings from the compiler. To ensure this, we are rejecting any build warnings in our continuous integration.

Documentation of this change can be added once #706 (or similar) is merged. Refer to [IS-739](https://jira.hyperledger.org/browse/IS-739) for this change.

### CODE REVIEW

**Comment by stevengubler** [comment]
As a sidenote, I approved this change with Nathan George, so let's please involve him on any discussion on whether or not this should be our strategy to comply with CII standards.

File: cli/src/main.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// If compiling with feature "fatal_warnings", the build will fail on all warnings
```
Comment (jovfer) [inline cli/src/main.rs]: I'm not sure, what this line is the best place for this comment. May be we should move it to `Cargo.toml` file?
@vimmerru ^

File: Jenkinsfile.ci [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!groovy
 
+def features = "--features fatal_warnings"
+def buildCommand = "cargo build ${features}"
```
[the remark this answers is not in the document]
Comment (stevengubler) [reply Jenkinsfile.ci]: `buildCommand` was initially a replacement for all entries, but when I pulled master, I had to change it up a little bit in order to get it to work with 
```groovy
def libindyFeatures = "--features sodium_static"
...
sh "cargo build $libindyFeatures"
```

I guess I'm not sure how to reconcile the specific needs of this specific build case and still abstract out our build command. Should I just remove the variable entirely, or is there a better way to do this?

File: cli/src/main.rs (hunk 1 above)
Comment (stevengubler) [reply cli/src/main.rs]: I like that idea. Makes a lot more sense. Should I make the change or wait for @vimmerru's input?

**Comment by stevengubler** [comment]
Alright, I made the changes you asked for. Anything else?

**Comment by jovfer** [comment]
@vimmerru @Artemkaaas @KitHat please remove warnings in current master and merge this PR

**Pull request merged**